packages/ui/src/views/layouts/main-content-layout.tsx

```diff
@@ -1,49 +1,57 @@
-import { CSSProperties, PropsWithChildren, ReactNode } from 'react'
+import { PropsWithChildren } from 'react'
 
+import { SidebarContext } from '@/components'
 import { useTheme } from '@/context'
-import { cn, wrapConditionalObjectElement } from '@/utils'
-import { ScrollArea, ScrollAreaProps } from '@components/scroll-area'
+import { cn } from '@/utils'
+import { AppBreadcrumbs, AppBreadcrumbsProps } from '@components/app-breadcrumbs'
 
-interface MainContentLayoutProps extends PropsWithChildren<unknown> {
-  className?: string
-  breadcrumbs?: ReactNode
-}
+const HalfArch = ({ className }: { className?: string }) => (
+  <div className="relative size-1.5 overflow-hidden">
+    <div
+      className={cn(
+        'absolute top-0 size-3 rounded-full border shadow-[0_0_0_20px_hsl(var(--canary-sidebar-background-01))]',
+        className
+      )}
+    />
+  </div>
+)
 
-export function MainContentLayout({ className, children, breadcrumbs }: MainContentLayoutProps) {
+type MainContentLayoutProps = PropsWithChildren<AppBreadcrumbsProps & { useSidebar?: () => SidebarContext }>
+
+export function MainContentLayout({ children, useSidebar, ...breadcrumbsProps }: MainContentLayoutProps) {
   const { isInset } = useTheme()
+  const sidebarData = useSidebar?.()
+  const isMobile = sidebarData?.isMobile
+  const withBreadcrumbs = breadcrumbsProps.breadcrumbs?.length > 0
 
   return (
-    <div className={cn('h-screen', className)}>
-      {!!breadcrumbs && (
-        <div
-          className={cn('bg-cn-background-1 sticky top-0 focus-within:z-20', { 'bg-sidebar-background-1': isInset })}
-        >
-          {breadcrumbs}
-        </div>
-      )}
-      <div className={cn('main-content-height', { 'overflow-hidden bg-sidebar-background-1 pb-1.5': isInset })}>
-        <ScrollArea
-          type="always"
-          orientation="both"
-          className={cn('h-full', { 'rounded-md border': isInset })}
-          viewportClassName={cn('!overflow-scroll min-h-full [&>div]:!grid [&>div]:min-h-full', {
-            'min-h-[calc(100%-6px)]': isInset
-          })}
-          scrollBarProps={{ vertical: { className: 'z-20' } }}
-          {...wrapConditionalObjectElement<ScrollAreaProps>(
-            {
-              scrollBarProps: { vertical: { className: 'h-auto z-20' }, horizontal: { className: '!bottom-1.5 z-20' } },
-              style: {
-                '--radix-scroll-area-corner-height': '6px',
-                '--radix-scroll-area-corner-width': '14px'
-              } as CSSProperties
-            },
-            isInset
-          )}
-        >
-          <div className={cn({ 'rounded-md bg-cds-background-1 min-w-fit': isInset })}>{children}</div>
-        </ScrollArea>
+    <>
+      <AppBreadcrumbs isMobile={isMobile} {...breadcrumbsProps} />
+      <div
+        className={cn(
+          'min-h-screen bg-cn-background-1',
+          { 'min-h-[calc(100vh-55px)]': withBreadcrumbs },
```

Comment:
We are using `55px` as a static height in many places. Can we refer it from a common css variable?

Reply:
Good idea
Should I add it to `packages/ui/src/styles.css`?
Something like this:
```css
:root, :host {
  --breadcrumbs-height: 55px;
}
```

cc @knagurski